README.md

```diff
@@ -676,7 +676,6 @@ Readme: https://github.com/extremeshok/clamav-unofficial-sigs
 #### Enable clamav-unofficial-sigs
 
 Create your `user.conf` file under `/mnt/docker/mail/clamav-unofficial-sigs` directory to configure clamav-unofficial-sigs updater. This file override the default configuration specified in [os.conf](https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/os.conf) and [master.conf](https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/master.conf). Don't forget, once you have completed the configuration of this file, set the value of `user_configuration_complete` to `yes` otherwise the script will not be able to execute.
```

Comment:
There seems to be an inconsistency:  `etc/clamav/unofficial-sigs` here vs `etc/clamav-unofficial-sigs` elsewhere, why was this changed?

Reply:
i just noticed it had changed when launching the clamav-unofficial-sigs.sh script, it seems a long tech debt, as even in the 5.6.2 official release it's already `/etc/clamav-unofficial-sigs` see https://github.com/extremeshok/clamav-unofficial-sigs/releases/tag/5.6.2 , but https://github.com/hardware/mailserver/blob/master/rootfs/usr/local/bin/clamav-unofficial-sigs.sh#L1293 :'( 

even is same script info : 

```
# Script Info
script_version="5.6.2"
script_version_date="2017-03-19"
minimum_required_config_version="72"
minimum_yara_clamav_version="0.99"
```

Reply:
Ah, so it was changed on their side, fair enough. Could you please update the readme.md to reflect the new path then? Many thanks!

Reply:
it seems changed now in "Files changed" can you confirm and resolve?

Reply:
Thank you, I just searched the repo for `clamav/unofficial` and there are two more occurrences, one in Makefile and one in os.conf, can you check it out please?

Reply:
@AndrewSav the one in Makefile seems accurate, as is it the link to test folder : 

`		-v "`pwd`/test/share/clamav/unofficial-sigs/user.conf":/var/mail/clamav-unofficial-sigs/user.conf \`


for the one in os.conf that are the specific points i wanted you to check/validate/correct as it relates to the todo list i did write on the desc of the MR

Reply:
@diroots hello, thank you for this. For `Makefile`: don't you think that for consistency we should rename the test folder too?

For `os.conf` can you share a bit more of why you are not sure? To me it seems that leaving in there a path that was changed everywhere else does not make sense, although I could miss something. Why do you think it might make sense?

Reply:
@AndrewSav for Makefile, could be good to rename the testfolder too. will it break any test? 

for os.conf : the references to old path are comments I did let on file for you, maintainers of  mailserver app, to review. as i'm not sure of what should be final definition, let me paste here to exchange on it : 


```
# ### https://github.com/mailserver2/mailserver specifics : ###
# taken from https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/os.conf#L7
# as service handler is s6-supervise

clamd_reload_opt="s6-svc -2 /services/clamd"
# this is in case the needed/expected keyword is now restart and not reload
clamd_restart_opt="s6-svc -2 /services/clamd"


# this is enabled here https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/os.conf#L6
clamd_socket="/run/clamav/clamd.ctl"
```

in your old os.conf file, the clamd_reload_opt was used, and modified to use s6-svc with signal 2 which is a SIGUSR2 signal. SIGUSR2 performs a Reload the signature databases. as per https://manpages.debian.org/bullseye/clamav-daemon/clamd.8.en.html#SIGUSR2


in latest os.debian.conf on the clamav-unofficiel-sigs project, https://github.com/extremeshok/clamav-unofficial-sigs/blob/master/config/os/os.debian.conf#L33 they do define a clamd_restart_opt value.

what is the s6svc command to perform a proper clamav restart? i did try s6-svc -r /services/clamd but clamd process do not respawn correctly. (taken from https://skarnet.org/software/s6/s6-svc.html) 

this is after a `s6-svc -r /services/clamd` or even after a `s6-svc -d /services/clamd && s6-svc -u /services/clamd`
Tue Nov 29 14:37:09 2022 -> --- Stopped at Tue Nov 29 14:37:09 2022
Tue Nov 29 14:37:09 2022 -> Socket file removed.

i need to force a single `s6-svc -u /services/clamd` to see clamd restart correctly
Tue Nov 29 14:38:55 2022 -> +++ Started at Tue Nov 29 14:38:55 2022
Tue Nov 29 14:38:55 2022 -> Received 0 file descriptor(s) from systemd.
Tue Nov 29 14:38:55 2022 -> clamd daemon 0.103.3 (OS: linux-gnu, ARCH: x86_64, CPU: x86_64)


also for last comment about the socket. i'm not sure if the socket is linked to clamav as i don't find any reference to it in clamav config files,... so do we need to keep it? 

and last info about os.conf 

in your original version, there were these 2 additional params : 

```
min_sleep_time="30"
max_sleep_time="60"
```

do we need to keep them or are these useless?

Reply:
@diroots 
> for Makefile, could be good to rename the testfolder too. will it break any test?

That would be good to check out, this is what the tests are for: we need to run them after any changes to make sure they do not break, I'd say it should be a pre-requisite for a successful merge. I would run them locally (a gotcha is that they cannot be run as root, I wasted a fair bit of time before realising that), but the PR pipeline can run them too, I approved it now for a first time contributor, all subsequent ones should not need an approval. When you ensure that tests work after that change please report back, or if they break and you cannot figure out why, let me know and I'll have a look.

> what is the s6svc command to perform a proper clamav restart? i did try s6-svc -r /services/clamd but clamd process do not respawn correctly. 

It's a tricky one. There is a [finish script](https://github.com/mailserver2/mailserver/blob/71688532934217c117d79e80b89b2bd36eeadefc/rootfs/usr/local/bin/setup.sh#L814) that is added to each service. It is intended to to stop restarting services if they restart too many times. but the side effect is that if a service finish with 0 exit code, `s6-svc -d` is called which will then prevent for the service to come up. Since the same finish script is used for all the services, not just for clamd I would imagine that the intention was that some of those are _meant_ to finish eventually and permanently. I would question why they are excuted with s6 then, but I did not design this so I can only guess. 

One workaround that I came up with, although it is not ideal, is to create a `down-signal` file in `/services/clamd` which contains a single line saying `SIGKILL`. It will make ` s6-svc -r /services/clamd` send `SIGKILL` instead of `SIGTERM` so the `finish` script will receive `256 9` instead of exit code of 0, and restart will proceed as planned. This is not ideal, because SIGTERM is obviously a more gentle approach, but without good understanding why exactly we have the `finish` script as is for all the services it may require more testing to come up with a different solution. If no one has a better idea and there are no visible drawbacks `SIGTERM` may work for us.

> also for last comment about the socket. i'm not sure if the socket is linked to clamav as i don't find any reference to it in clamav config files,... so do we need to keep it?

Sorry, which socket is that, can you give some more info?

>do we need to keep them or are these useless?

Why do you think they are useless? the are different from [the defaults](https://github.com/extremeshok/clamav-unofficial-sigs/blob/4a5b8f1ef64a6f1f744c98b605d7ed83ce5c7453/config/master.conf#L572) so to me it looks like they serve some purpose? I do not mind removing them since they do not seem to matter much, the defaults seem sensible, but on the other hand some one added them there with some purpose which we do not know, so may be it is safer to leave them in.

Reply:
> Sorry, which socket is that, can you give some more info?

this socket : 
https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/os.conf#L6

> I do not mind removing them since they do not seem to matter much, the defaults seem sensible, but on the other hand some one added them there with some purpose which we do not know, so may be it is safer to leave them in

these timeouts help during the time randomization of signatures updates ran by cron. 
in original, it let between 1 and 10 minutes. in this setup, it narrow this between 30s and one minut. 

> # Enable or disable download time randomization.  This allows the script to
> # be executed via cron, but the actual database file checking will pause
> # for a random number of seconds between the "min" and "max" time settings
> # specified below.  This helps to more evenly distribute load on the host
> # download sites.

bigger timeframe helps to more evenly distribute load on the host download sites. so narrowing the sleep times generate more requests more often on download sites. 
moreover signatures do not need to be updated so often. and some signature host sites warn you can be blacklisted if you do request too often. i would let default values there. (it's what i did on our setup, and it do not seems to cause problems as per the logs.

Reply:
SIGKILL is arsh and can lead to ongoing clamd executions being stopped while running. 

from there https://github.com/extremeshok/clamav-unofficial-sigs/blob/master/clamav-unofficial-sigs.sh#L1336 it seems restart is fired only if reload is not fully working. in your previous implementations, you were relying only on reload only, as per your previous os.conf file. should we stay as is ? whatever the script will log if reload fails and restart not implemented

Reply:
@AndrewSav if you think we can rethink the startup process to improve it in a separate PR?

Reply:
>also for last comment about the socket. i'm not sure if the socket is linked to clamav as i don't find any reference to it in clamav config files,... so do we need to keep it?
>this socket : https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/os.conf#L6

It is mentioned here: https://github.com/mailserver2/mailserver/blob/master/rootfs/etc/clamav/unofficial-sigs/master.conf#L423 and if I'm not mistaken it is still there with the latest version in the clamav unofficial sigs repo. Do you think they forgot to remove it from there by mistake and it is not longer used? If so may be worth opening issue with them to clarify. Otherwise for the purpose of merging the PR I'd probably leave it there on the off chance we are missing it's usage.

> bigger timeframe helps to more evenly distribute load on the host download sites. so narrowing the sleep times generate more requests more often on download sites.
>moreover signatures do not need to be updated so often. and some signature host sites warn you can be blacklisted if you do request too often. i would let default values there. (it's what i did on our setup, and it do not seems to cause problems as per the logs.

Ok, that's sounds reasonable, let's remove them then.

> from there https://github.com/extremeshok/clamav-unofficial-sigs/blob/master/clamav-unofficial-sigs.sh#L1336 it seems restart is fired only if reload is not fully working. in your previous implementations, you were relying only on reload only, as per your previous os.conf file. should we stay as is ? whatever the script will log if reload fails and restart not implemented

If reload works, then leaving as it is is the preferred option, but from what you wrote earlier I thought that they no longer using reload and using restart instead. If that's not the case and reload still works as it used to, let's stick with how it used to work.

---

In order to merge the PR we want: 
- change to the test folder path for clamav to make it consistent across the board and fix broken tests if any
- remove the `clamd_restart_opt` option since it's not working (if reload will still work - otherwise we need to think of something else)
- remove `min_sleep_time` and `max_sleep_time` since it looks like the defaults are better

Reply:
ok @AndrewSav see https://github.com/diroots/mailserver/commit/12fe85b896879d5a45d709240e291126184a7a05